src/Terrain/Maps/MapReader.java

```diff
@@ -0,0 +1,174 @@
+package Terrain.Maps;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import Engine.XYCoord;
+import Terrain.Environment;
+import Terrain.MapInfo;
+
+public class MapReader extends IMapBuilder
+{
+  /**
+   * Tells the MapReader to read in the maps.
+   */
+  public static ArrayList<MapInfo> readMapData()
+  {
+    ArrayList<MapInfo> importMaps = new ArrayList<MapInfo>();
+
+    // This try{} is to safeguard us from exceptions if the res/map folder doesn't exist.
+    // If it fails, we don't need to do anything in the catch{} since we just won't have anything in our list.
+    try
+    {
+      final File folder = new File("res/map");
+
+      for( final File fileEntry : folder.listFiles() )
+      {
+        // we aren't checking subdirectories, yet
+        if( !fileEntry.isDirectory() )
+        {
+          try
+          {
+            // We get the filename, and make it look nice for our map list.
+            String mapName = fileEntry.getName();
+            // Mostly, we just don't want to try to interpret the python script as a map. That'd be bad.
+            if( !mapName.contains(".map") )
+              continue;
+            // underscores->spaces makes it pretty
+            mapName = mapName.replaceAll("_", " ");
+            mapName = mapName.replaceAll(".map", "");
+            System.out.println("INFO: Parsing map: " + mapName);
+
+            // We need a list of who starts owning what properties. This is that list.
+            // Each arraylist contains coordinates, and which list it is denotes who owns that property.
+            ArrayList<ArrayList<XYCoord>> properties = new ArrayList<ArrayList<XYCoord>>();
+            // Will support 14 sides in the future, possibly more.
+            for( int i = 0; i < 4; i++ ) // TODO: add more sides
+            {
+              properties.add(new ArrayList<XYCoord>());
+            }
+
+            Scanner scanner = new Scanner(fileEntry);
+            // We need the first line so we can pre-fill our terrain data array with the proper number of sub-arrays.
+            String line = scanner.nextLine();
+
+            // The game's internal representation of the map is transposed from what we see.
```

Comment:
I think this comment could be misleading. If you manually write out the 2D terain array, then yes, it appears to be transposed from what is drawn. However, if you pick an x/y coordinate on the map, it <i>does</i> correspond to the tile at that [x][y] position in the array. The transposition thing is just a function of how it is necessary to write 2D arrays in code, and only really comes into play when e.g., you try to look at one of our in-code maps. The remaining explanation is useful though.